### ISSUE TITLE
Partial cache invalidation not working / how does Turpentine interact with cachetags?

### ISSUE THREAD

**Issue by Nemykal** [nexcess/magento-turpentine#568]
Hi,

Firstly, thank you for making this excellent plugin. I am a systems administrator who works with a lot of Magento installations and we've been able to make customers sites scale and survive heavy load thanks to the Varnish integration. 

I have some questions about partial cache invalidation with Turpentine. We have a customer who wants very long TTLs but also wants to be able to make and see changes to their products throughout the day. All of my tests have been with Magento EE 1.13.

The scenario:

1. A Magento site sells widgets and has a "Foobar" product that they want to update the price of.
1. The cache is hot. Not only for the product page itself, but also for the category pages where this product appears. In fact, this "Foobar" product has shown up in several search result pages
1. In the Magento admin backend, they update the product and click save.
1. **The problem:** The associated category and search result pages aren't flushed

During my testing, I found that the product page itself is indeed correctly flushed, but other "dirtied" pages are not.

Note: I also tried re-indexing and refreshing all the caches, and the old content was still there.
Yes, I am refreshing with ctrl f5. The Magento enterprise FPC is off. No luck :(

I am not a PHP or Magento developer myself so I don't know whether this issue needs to be addressed in the Turpentine plugin itself or by this Magento site's developers or if it's a problem internal to Magento. Please let me know if any of my assumptions are incorrect. 

From my time spent working with Magento, I think the concept I'm trying to ask about here is "cache tags". Taking varnish out of the picture entirely for a moment, consider a Magento site with Redis full page cache or Memcache (or Files, or Database etc) configured in local.xml. Apparently one of the advantages of CM/Mage Redis in Magento is that it supports "cache tags". And that one of the disadvantages of Memcache in Magento is that it does not, so it has to maintain a table in the database [1] which is just awful. 

[1] https://www.sonassi.com/knowledge-base/magento-kb/what-is-memcache-actually-caching-in-magento/

So basically:

1. Are we doing something wrong? Should turpentine already support cache tags / partial cache invalidation like I described? 
1. If not, could something like this please be added to Turpentine somehow? Or if there's another way to achieve this... 

Also: does anyone know of a good way to monitor traffic on Varnish's management port 6082? I want to see what turpentine is actually telling Varnish to purge

Cheers, please let me know if you need more information etc,
Michael.

**Comment by miguelbalparda** [comment]
@Nemykal today we merged #522 to devel,it should help you flushing only certain pages manually.
